Gemfile

```diff
@@ -11,6 +11,7 @@ gemspec
 gem "hocon", '>= 1.2.5'
 gem "json-schema", '>= 2.8.0'
 gem "puma", '>= 3.12.0'
+gem "puppet", git: "https://github.com/donoghuc/puppet.git", branch: 'GH-1152'
```

Comment:
Instead of trying to catch and munge erros from loading task metadata, some additional error handling was implemented upstream. https://github.com/puppetlabs/puppet/pull/7769

Reply:
https://tickets.puppetlabs.com/browse/PUP-10095

Reply:
Heh, should've merged that right when you put it up and I think it could've made it into the release tomorrow :) As it is, I think we can get the gem from the puppetlabs master branch until it's released (which I don't think is even scheduled, since 6.4.4 hasn't gone out yet)

Reply:
The nice thing about this is that nothing is *dependent* on the change. Whenever we pick up a version of puppet with the improved messaging it will "just work". So all I need to do is delete this line and we are good. ( i guess I need a changelog too...)